fix stof error

JunseoMin opened the pull request:
Hi , @LimHyungTae 

Thank you for all your amazing contributions! I’ve been following your work on GitHub, LinkedIn, and through your papers.

While running ORORA with the Mulran dataset (sequence DDC01), I encountered an error in the `loadGTPoses` function within `include/SE2posemanager.hpp`. The error was:

```
terminate called after throwing an instance of 'std::out_of_range' what(): stof
```
After some investigation, I discovered that the issue occurs because the `radar_odometry.csv` file contains the value `1.43493e-42`. To handle this properly, I changed the conversion from `stof` to `stod`.

Additionally, I noticed that the vector `parsed_xyyaw` was mistakenly parsing the yaw value twice (using `words[3]` for both y and yaw). I corrected it so that it now properly reads x, y, and yaw:

```
std::vector<float> parsed_xyyaw = {stod(words[1]), stod(words[2]), stod(words[3])};
```
I’ve also added a few safety functions to improve robustness.

Thank you for taking the time to review these changes. I’m really excited to contribute to a codebase by someone I truly admire!

Best regards,
Junseo Min

JunseoMin committed:
fix stof error

LimHyungTae commented on include/SE2posemanager.hpp:
```diff
@@ -22,14 +22,24 @@ void loadGTPoses(const std::string &gt_path, std::unordered_map<int64, std::vect
+    // std::getline(ifs, line);
+
     while (std::getline(ifs, line)) {
+        if (line.empty()) continue;
```
Does this situation occur in the current pipeline?

JunseoMin replied:
Nope. This situation does not occur in the current pipeline.

LimHyungTae commented on include/SE2posemanager.hpp:
```diff
@@ -22,14 +22,24 @@ void loadGTPoses(const std::string &gt_path, std::unordered_map<int64, std::vect
+
     while (std::getline(ifs, line)) {
+        if (line.empty()) continue;
+        float x,y,yaw;
```
Why do we need to declare unused variables here?

JunseoMin replied:
These two lines (28,29) were only used for debugging, and I accidentally forgot to remove them. Thanks for checking!

LimHyungTae commented on include/SE2posemanager.hpp:
```diff
@@ -22,14 +22,24 @@ void loadGTPoses(const std::string &gt_path, std::unordered_map<int64, std::vect
+
+        if (words.size() < 4) {
+            std::cerr << "Skipping invalid line: " << line << std::endl;
+            continue;
```
Is it okay to use continue? Wouldn't it be better to throw std::logic_error and provide guidance to the user on what went wrong?

LimHyungTae reviewed (commented):
@JunseoMin Oh man appericate you having your warm and kind attention! 

Could you consider my comments and push a commit again?

JunseoMin committed:
add error log

LimHyungTae commented:
Again, thanks for your attention, @JunseoMin! By any chance, do you plan to use ORORA in ROS2 in your lab? If so, I'm always open to discussing it!

JunseoMin commented:
Thanks for reviewing! I'm interested in radar single-shot localization, but I need to discuss it with my professor first. (I'm currently doing an internship.) If I start researching with ORORA, can I reach out to you via email?

JunseoMin committed:
chore run_orora.launch

JunseoMin commented:
As a kind of fan of yours, I really appreciate your response!

For summery:
1. I removed lines 28 and 29, and no empty lines appeared in the current pipeline.
2. I added a logic error message: 'Invalid line detected from GT dataset!'
